### PR TITLE
File manager

### PR DESCRIPTION
**Pull request by belspirit**
# File Manager

Task: https://github.com/AlreadyBored/nodejs-assignments/blob/main/assignments/file-manager/assignment.md

Done: 25.06.2023 / deadline 26.06.2023

How to check a student's project: https://github.com/AlreadyBored/nodejs-assignments/blob/main/assignments/file-manager/score.md

PS: It is possible for the **os** command to pass multiple flags in one call. For example:
```bash
os --architecture --EOL --homedir --cpus --username
```

Discord : Sergey Labetik (@belspirit)
If you have any remarks or questions, please write to me.

Score: 330 / 330

## Basic Scope
- General
    - [x] **+6** Application accepts username and prints proper message
    - [x] **+10** Application exits if user pressed `ctrl+c` or sent `.exit` command and proper message is printed
- Operations fail
    - [x] **+20** Attempts to perform an operation on a non-existent file or work on a non-existent path result in the operation fail
    - [x] **+10** Operation fail doesn't crash application
- Navigation & working directory operations implemented properly
    - [x] **+10** Go upper from current directory
    - [x] **+10** Go to dedicated folder from current directory
    - [x] **+20** List all files and folders in current directory
- Basic operations with files implemented properly
    - [x] **+10** Read file and print it's content in console
    - [x] **+10** Create empty file
    - [x] **+10** Rename file
    - [x] **+10** Copy file
    - [x] **+10** Move file
    - [x] **+10** Delete file
- Operating system info (prints following information in console) implemented properly
    - [x] **+6** Get EOL (default system End-Of-Line)
    - [x] **+10** Get host machine CPUs info (overall amount of CPUS plus model and clock rate (in GHz) for each of them)
    - [x] **+6** Get home directory
    - [x] **+6** Get current *system user name* (Do not confuse with the username that is set when the application starts)
    - [x] **+6** Get CPU architecture for which Node.js binary has compiled
- Hash calculation implemented properly
    - [x] **+20** Calculate hash for file 
- Compress and decompress operations
    - [x] **+20** Compress file (using Brotli algorithm)
    - [x] **+20** Decompress file (using Brotli algorithm)

## Advanced Scope

- [x] **+30** All operations marked as to be implemented using certain streams should be performed using Streams API
- [x] **+20** No synchronous Node.js API with asynchronous analogues is used (e.g. not used `readFileSync` instead of `readFile`)  
- [x] **+20** Codebase is written in ESM modules instead of CommonJS
- [x] **+20** Codebase is separated (at least 7 modules)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  args.map((arg) => {
+    if (arg.startsWith("--username=")) {
+      [, username] = arg?.split("=");
+    }
+  });
+  console.log(colors.fgGreen, `Welcome to the File Manager, ${username}!`);
+
+  whereIam();
+
+  // workarround for the Windows CTRL+C hotkey
```
Comment (RamanHupalenka) [inline src/index.js]: Just for your information, this "workaround" broke your solution.

Comment (belspirit) [reply src/index.js]: I didn't test it under Windows... Just relied on the StackOverflow. Thank you for the information.

Comment (belspirit) [reply src/index.js]: Could you test it under linux/iOs please?

Comment (RamanHupalenka) [reply src/index.js]: I don't have Linux/iOS.

Comment (RamanHupalenka) [reply src/index.js]: > I didn't test it under Windows... Just relied on the StackOverflow. Thank you for the information.

Never rely on StackOverflow answers that were given >8 years ago 😉

Comment (RamanHupalenka) [reply src/index.js]: https://nodejs.org/docs/latest-v18.x/api/process.html#signal-events

Comment (belspirit) [reply src/index.js]: Please, review it again. I just fixed this Windows `stdin` "workaround".

Comment (RamanHupalenka) [reply src/index.js]: You really want -30% from your total score?

Comment (belspirit) [reply src/index.js]: I don't. But 220 is more than 101, correct?

Comment (belspirit) [reply src/index.js]: I mean 231

Comment (belspirit) [reply src/index.js]: Btw, on the real projects we use WSL to run node. And Docker desktop

Comment (RamanHupalenka) [reply src/index.js]: Okay, as you wish, I'll review it again. The result will be known no later than Friday 😉

**Review by RamanHupalenka — APPROVED** [review]

**Pull request merged**